docs/tutorials/extending_admin_ui/4_filtering_the_content_list.md

```diff
@@ -0,0 +1,210 @@
+
+``` php
+if ($contentTypeIdentifier) {
+    $criterions[] = new Criterion\ContentTypeIdentifier($contentTypeIdentifier);
+}
+```
+
+After the lines setting the `$paginator` parameters, add the following code block:
+
+``` php
+$contentTypes = [];
+$contentTypeGroups = $this->contentTypeService->loadContentTypeGroups();
+foreach ($contentTypeGroups as $group) {
+    $contentTypes[$group->identifier] = $this->contentTypeService->loadContentTypes($group);
+}
+```
+
+Finally, provide the new parameter to `$this->render`, after `articles`:
+
+``` php
+'contentTypes' => $contentTypes,
+```
+
+??? tip "Complete controller code"
+
+    ``` php hl_lines="25 33 34 35 45 46 47 48 49 54"
+    <?php
+
+    namespace EzSystems\ExtendingTutorialBundle\Controller;
+
+    use EzSystems\EzPlatformAdminUiBundle\Controller\Controller;
+    use eZ\Publish\API\Repository\SearchService;
+    use eZ\Publish\API\Repository\ContentTypeService;
+    use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+    use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+    use eZ\Publish\Core\Pagination\Pagerfanta\LocationSearchAdapter;
+    use Pagerfanta\Pagerfanta;
+
+    class AllContentListController extends Controller
+    {
+        private $searchService;
+
+        private $contentTypeService;
+
+        public function __construct(SearchService $searchService, ContentTypeService $contentTypeService)
+        {
+            $this->searchService = $searchService;
+            $this->contentTypeService = $contentTypeService;
+        }
+
+        public function listAction($contentTypeIdentifier = false, $page = 1)
+        {
+            $query = new LocationQuery();
+
+            $criterions = [
+                new Criterion\Visibility(Criterion\Visibility::VISIBLE),
+            ];
+
+            if ($contentTypeIdentifier) {
+                $criterions[] = new Criterion\ContentTypeIdentifier($contentTypeIdentifier);
+            }
+
+            $query->query = new Criterion\LogicalAnd($criterions);
+
+            $paginator = new Pagerfanta(
+                new LocationSearchAdapter($query, $this->searchService)
+            );
+            $paginator->setMaxPerPage(8);
+            $paginator->setCurrentPage($page);
+
+            $contentTypes = [];
+            $contentTypeGroups = $this->contentTypeService->loadContentTypeGroups();
+            foreach ($contentTypeGroups as $group) {
+                $contentTypes[$group->identifier] = $this->contentTypeService->loadContentTypes($group);
+            }
+
+            return $this->render('@EzSystemsExtendingTutorial/list/all_content_list.html.twig', [
+                'totalCount' => $paginator->getNbResults(),
+                'articles' => $paginator,
+                'contentTypes' => $contentTypes,
+            ]);
+        }
+    }
+    ```
+
+## Change the template
+
+The last thing to do is to update the template by adding a drop-down menu for choosing Content Types.
```

Comment:
Same here, path and name can be mentioned: `src/EzSystems/ExtendingTutorialBundle/Resources/views/list/all_content_list.html.twig`